processor/redactionprocessor/processor.go

```diff
@@ -113,15 +113,20 @@ func (s *redaction) processAttrs(_ context.Context, attributes pcommon.Map) {
 
 		// Mask any blocked values for the other attributes
 		strVal := value.Str()
+		toBlockUnqiue := make(map[string]struct{})
 		for _, compiledRE := range s.blockRegexList {
 			match := compiledRE.MatchString(strVal)
 			if match {
-				toBlock = append(toBlock, k)
+				toBlockUnqiue[k] = struct{}{}
```

Comment:
I wonder if `sync.Once` would be a little more efficient than the new map and second for loop.

Reply:
Ah, I didn't notice it's inside a `attributes.Range` loop and `k` never changes here . Then I think a simple bool check should be even better.